lua/nvim-lsp-installer/ui/display.lua

```diff
@@ -99,42 +99,72 @@ end
     local namespace = vim.api.nvim_create_namespace(("lsp_installer_%s"):format(name))
-    local buf, renderer, mutate_state, get_state, unsubscribe
+    local buf_handle, renderer, mutate_state, get_state, unsubscribe, win_id
     local has_initiated = false
 
     local function open(opts)
         opts = opts or {}
         local win_width, highlight_groups = opts.win_width, opts.highlight_groups
+        opts.layout = opts.layout or "popup"
 
-        if win_width then
-            vim.cmd(("%dvnew"):format(win_width))
+        if opts.layout == "popup" then
+            local win_height = vim.o.lines - vim.o.cmdheight - 2 -- Add margin for status and buffer line
+            win_width = vim.o.columns
+            local popup_layout = {
+                relative = "editor",
+                height = math.floor(win_height * 0.9),
+                width = math.floor(win_width * 0.8),
+                style = "minimal",
+                border = "rounded",
+            }
+            popup_layout.row = math.floor((win_height - popup_layout.height) / 2)
+            popup_layout.col = math.floor((win_width - popup_layout.width) / 2)
+            buf_handle = vim.api.nvim_create_buf(false, true)
+            win_id = vim.api.nvim_open_win(buf_handle, true, popup_layout)
+            vim.lsp.util.close_preview_autocmd({ "BufHidden", "BufLeave" }, win_id)
         else
-            vim.cmd [[vnew]]
+            if win_width then
+                vim.cmd(("%dvnew"):format(win_width))
+            else
+                vim.cmd [[vnew]]
+            end
+            win_id = vim.api.nvim_get_current_win()
+            buf_handle = vim.api.nvim_get_current_buf()
         end
 
-        local win = vim.api.nvim_get_current_win()
-        buf = vim.api.nvim_get_current_buf()
-
-        vim.api.nvim_buf_set_option(buf, "modifiable", false)
-        vim.api.nvim_buf_set_option(buf, "swapfile", false)
-        vim.api.nvim_buf_set_option(buf, "textwidth", 0)
-        vim.api.nvim_buf_set_option(buf, "buftype", "nofile")
-        vim.api.nvim_buf_set_option(buf, "bufhidden", "wipe")
-        vim.api.nvim_buf_set_option(buf, "buflisted", false)
-        vim.api.nvim_buf_set_option(buf, "filetype", "lsp-installer")
-
-        vim.api.nvim_win_set_option(win, "wrap", false)
-        vim.api.nvim_win_set_option(win, "spell", false)
-        vim.api.nvim_win_set_option(win, "number", false)
-        vim.api.nvim_win_set_option(win, "relativenumber", false)
-        vim.api.nvim_win_set_option(win, "foldenable", false)
-        vim.api.nvim_win_set_option(win, "signcolumn", "no")
-        vim.api.nvim_win_set_option(win, "colorcolumn", "")
+        local buf_opts = {
+            modifiable = false,
+            swapfile = false,
+            textwidth = 0,
+            buftype = "nofile",
+            bufhidden = "wipe",
+            buflisted = false,
+            filetype = "lsp-installer",
+        }
+
+        local win_opts = {
+            number = false,
```

Comment:
I believe we do need to set `norelativeumber` too, they're not mutually exclusive

Reply:
TIL!